### PR TITLE
[new-feature] Can require php version or extensions

### PR DESCRIPTION
**Pull request by ArthurHlt**
This PR give the possibility to do this:
```php
<?php
<<<CONFIG
packages:
    - "php: >=5.5.0"
    - "twig/twig: ~1.0"
    - "ext-soap: *"
CONFIG;
//...
```
Like this it can say to user to use the good php version and good extension that's a gist intend to work with.
I don't know if it's should be documented cause it's a default composer feature.

### CODE REVIEW

**Comment by ArthurHlt** [comment]
Sorry when adding test I didn't see that my IDE move methods in `src/SensioLabs/Melody/Tests/IntegrationTest.php`. Just the method `testRunWithConstraints` was added.

File: src/SensioLabs/Melody/Configuration/ConfigurationParser.php [hunk 1]
```diff
@@ -12,7 +12,7 @@
 class ConfigurationParser
 {
     const PACKAGE_DELIMITER = ':';
-    const PACKAGE_REGEX = '/^(|[a-zA-Z0-9]([_.-]?[a-zA-Z0-9]+)*\\/[a-zA-Z0-9]([_.-]?[a-zA-Z0-9]+)*)$/';
+    const PACKAGE_REGEX = '/^((|[a-zA-Z0-9]([_.-]?[a-zA-Z0-9]+)*\\/[a-zA-Z0-9]([_.-]?[a-zA-Z0-9]+)*)|php|ext-[a-zA-Z0-9_.-]*)$/';
```
Comment (staabm) [inline src/SensioLabs/Melody/Configuration/ConfigurationParser.php]: After `ext-` it should contain at least one char `+` in the ext name instead of 0 `*`

**Comment by staabm** [comment]
:+1:

**Comment by ArthurHlt** [comment]
You're right, change made.

File: src/SensioLabs/Melody/Tests/IntegrationTest.php [hunk 2]
```diff
@@ -48,6 +48,13 @@ public function testRunWithCache()
         $this->assertSame('Hello world', $output);
     }
 
+    public function testRunWithConstraints()
+    {
+        $this->melodyRun('hello-world-with-constraints.php');
+        $output = $this->melodyRun('hello-world-with-constraints.php');
```
Comment (xabbuh) [inline src/SensioLabs/Melody/Tests/IntegrationTest.php]: Is it necessary to run `melodyRun()` twice?

File: src/SensioLabs/Melody/Tests/IntegrationTest.php [hunk 3]
```diff
@@ -146,7 +153,7 @@ private function melodyRun($fixture, array $options = array())
 
     private function cleanCache()
     {
-        $this->fs->remove(sys_get_temp_dir().'/melody');
+        $this->fs->remove(sys_get_temp_dir() . '/melody');
```
Comment (xabbuh) [inline src/SensioLabs/Melody/Tests/IntegrationTest.php]: Why that change?

Comment (ArthurHlt) [inline src/SensioLabs/Melody/Tests/IntegrationTest.php]: I don't know this must be from my IDE

Comment (ArthurHlt) [inline src/SensioLabs/Melody/Tests/IntegrationTest.php]: I don't know this must be from my IDE

File: src/SensioLabs/Melody/Tests/IntegrationTest.php (hunk 2 above)
Comment (ArthurHlt) [inline src/SensioLabs/Melody/Tests/IntegrationTest.php]: Suppose it was for gache testing, it's similar to: https://github.com/sensiolabs/melody/blob/master/src/SensioLabs/Melody/Tests/IntegrationTest.php#L44

Comment (staabm) [inline src/SensioLabs/Melody/Tests/IntegrationTest.php]: ok then please drop the duplicated invocation

**Comment by ArthurHlt** [comment]
I've removed the second `melodyRun` from `testWithConstraint`

File: src/SensioLabs/Melody/Tests/IntegrationTest.php (hunk 3 above)
Comment (lyrixx) [inline src/SensioLabs/Melody/Tests/IntegrationTest.php]: Can you revert this change ?

Comment (ArthurHlt) [inline src/SensioLabs/Melody/Tests/IntegrationTest.php]: done.

**Comment by lyrixx** [comment]
Thanks for your work on this new feature!
